### ISSUE TITLE
Story v2.0 - make stories break less

### ISSUE THREAD

**Issue by AnaBelgun** [TerriaJS/terriajs#5598]
Currently the stories created in v7 would easily break in v8 due to migration issues. 
The stories created in v8 would still break due to one or combination of reasons: 
- data service removed / decommissioned by the custodian 
- data service url modified by the custodian without decommissioning
- bugs due to latest software releases

Breaking stories create a very bad user experience which can be addressed: 
- making the sharelinks more reliable
- better handling the url changes 
- better handling of error messages (differentiation between data and software bugs)
- graceful transition to other data. scenes when data is decommissioned 
- better handling of re-arranging catalogue (when not decommissioning, removing datasets) - https://github.com/TerriaJS/terriajs/issues/5646

**Comment by wibily** [comment]
split into two tickets:
1) Warning message architecture dependent on #5459

**Comment by AnaBelgun** [comment]
This ticket is superseded by: 
#5459 - handling errors which addresses the handling of errors when they cannot be avoided nor controlled
#5646 - re-thinking catalogue data which has not been removed and how it affect share links
To avoid confusion, will close this ticket. 
Problem brief transferred to Epic - Story v2.0 https://github.com/terriajs/terriajs/issues/3532
